### PR TITLE
[GLUTEN-6026][VL] Add Support for HiveFileFormat parquet write for Spark 3.4+

### PR DESCRIPTION
**Pull request by surnaik**
## What changes were proposed in this pull request?

Adds support of HiveFileFormat write when output file type is of type parquet

(Fixes: #6026 )

## How was this patch tested?

Fixed the existing UTs to check whether pushed to Velox.
Checked manually in Spark.

### CODE REVIEW

**Comment by surnaik** [comment]
Please take a look - @ulysses-you , Thanks!!

File: backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala
Comment (ulysses-you) [inline backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala:229]: Can we add one more config to control if we transofrm hive file format ? We can enable it by defualt.

**Comment by ulysses-you** [comment]
thank you @surnaik , it seems tests did not pass, can you take a look ?

**Comment by surnaik** [comment]
> thank you @surnaik , it seems tests did not pass, can you take a look ?

This is strange, locally passing, checking the logs, it says nativeUsed was false, investigating further

**Comment by surnaik** [comment]
> thank you @surnaik , it seems tests did not pass, can you take a look ?

Found the issue, have raised the fix, waiting for CI to pass.

**Review by ulysses-you — APPROVED** [review]
lgtm if tests pass, thank you

**Comment by surnaik** [comment]
@ulysses-you could you please take a look once again, all the tests are now passing. Thanks for your time!

**Comment by surnaik** [comment]
@PHILO-HE could you please take a look as well. Thanks!

**Review by PHILO-HE — APPROVED** [review]
Looks nice! Thanks!

**Review by ulysses-you — APPROVED** [review]

**Pull request merged**
